src/gui/qgsmapcanvas.cpp

```diff
@@ -1010,7 +1010,7 @@ void QgsMapCanvas::zoomToSelected( QgsVectorLayer *layer )
     QgsFeatureIterator fit = layer->getFeatures( req );
     QgsFeature f;
     QgsPointXY closestPoint;
-    double closestSquaredDistance = extentRect.width() + extentRect.height();
+    double closestSquaredDistance = pow( extentRect.width() + extentRect.height(), 2.0 );
```

Comment:
shouldn't this be 

```suggestion
    double closestSquaredDistance = pow( extentRect.width(), 2.0 ) + pow( extentRect.height(), 2.0 );
```
?

Reply:
That's correct, it shouldn't change anything but let's be precise